### DISCUSSION TITLE
Consideration of YOLOv8 over OpenCV for Object Detection in Frigate

### DISCUSSION THREAD

**Discussion by DioCrafts** [blakeblackshear/frigate#12109] [Ideas]
Summary:
This discussion aims to explore the potential benefits and implications of adopting YOLOv8 for object detection in Frigate, as opposed to the current use of OpenCV.

Background:
Frigate currently leverages OpenCV for object detection, which has been instrumental in achieving robust and reliable performance. However, with the advancements in object detection models, particularly the introduction of YOLOv8, it's worth discussing whether transitioning to YOLOv8 could enhance the performance and accuracy of object detection in Frigate.

Reasons to Consider YOLOv8:

Improved Accuracy:

YOLOv8 offers state-of-the-art accuracy in object detection tasks. Numerous benchmarks indicate that YOLOv8 outperforms earlier models, including YOLOv5 and traditional methods employed by OpenCV, in terms of mean Average Precision (mAP).
Real-time Performance:

Like its predecessors, YOLOv8 is optimized for real-time object detection, providing high FPS (frames per second) with minimal latency. This makes it highly suitable for applications like Frigate that require real-time monitoring and alerting.
Better Handling of Small Objects:

YOLOv8 has shown significant improvements in detecting small objects, which is often a challenging scenario for models. This could enhance the capability of Frigate in scenarios involving smaller or more distant objects.
Advanced Features:

YOLOv8 includes features such as anchor-free detection and a more efficient backbone network, which could contribute to better overall performance and reduced computational load.
Community and Support:

The YOLO community is large and active, providing extensive documentation, tutorials, and community support. This can facilitate easier implementation and troubleshooting.
Potential Challenges:

Implementation Effort:

Transitioning to YOLOv8 would require significant changes in the codebase. This includes integrating the YOLOv8 model, adjusting the preprocessing and post-processing steps, and possibly retraining the model on relevant datasets.
Resource Requirements:

While YOLOv8 is efficient, it may still demand higher computational resources compared to traditional OpenCV methods, particularly on edge devices with limited capabilities.
Testing and Validation:

Extensive testing and validation would be necessary to ensure that YOLOv8 meets the performance standards and reliability required for production deployment in Frigate.
Discussion Points:

What are the specific pain points with the current OpenCV implementation that could be addressed by YOLOv8?
How significant are the accuracy and performance gains with YOLOv8 in practical use cases relevant to Frigate?
What are the potential trade-offs in terms of resource consumption and complexity?
How can we effectively manage the transition to YOLOv8 while minimizing disruptions to current functionalities?

**Comment by hawkeye217** [comment]
See https://github.com/blakeblackshear/frigate/pull/10717

**Comment by blakeblackshear** [comment]
Frigate doesn't use opencv for object detection. Also, we recently removed yolov8 support due to licensing issues: https://github.com/blakeblackshear/frigate/pull/10717

**Reply by DioCrafts** [reply]
Makes sense, thank you for your response!!

**Comment by aeozyalcin** [comment]
Hi, original dev who added yolox, yolov5 and yolov8 to frigate openvino (https://github.com/blakeblackshear/frigate/pull/5523) here. Just stopped by to say how disappointed I was to see the support being dropped for Frigate v14. I respect the repo owner's decisions, but I disagree with their interpretation of AGPL, and how it applies to Frigate - especially given Frigate doesn't include any models. I am hoping Frigate can quickly adopt a "3rd party add-on" model, so yolo inference code can be decoupled from the Frigate repo.

**Reply by philippedegeer** [reply]
Yes, same approach for me. The current YoloNAS support (in ver 14) is working very well but this is a fast moving field and it would be nice to be able to keep up with the latest developments without too much pain.
